torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,446 @@
+        sentence_eed = []
+        return scores, total_num_sentences, sentence_eed
+
+    # calculate score
+    scores = 0.0
+    total_num_sentences = 0.0
+
+    sentence_eed: Optional[List[Tensor]] = None
+    if return_sentence_level_score:
+        sentence_eed = []
+
+    for reference_words, hypothesis in zip(reference_corpus, hypothesis_corpus):
+        score, num_sentences = _compute_sentence_statistics(
+            reference_words, hypothesis, alpha, rho, deletion, insertion
+        )
+
+        if sentence_eed is not None:
+            sentence_eed.append(_eed_compute(tensor(scores), tensor(total_num_sentences)).unsqueeze(0))
+
+        scores += score
+        total_num_sentences += num_sentences
+
+    return tensor(scores), tensor(total_num_sentences), sentence_eed
+
+
+def eed(
+    reference_corpus: Sequence[Union[str, Sequence[str]]],
+    hypothesis_corpus: Union[str, Sequence[str]],
+    language: Literal["en", "ja"] = "en",
+    return_sentence_level_score: bool = False,
+    alpha: float = 2.0,
+    rho: float = 0.3,
+    deletion: float = 0.2,
+    insertion: float = 1.0,
+) -> Tensor:
+    """Computes extended edit distance score (`EED`_) [1] for strings or list of strings. The metric utilises the
+    Levenshtein distance and extends it by adding an additional jump operation.
+
+    Args:
+        reference_corpus:
+            An iterable of iterables of reference corpus.
+        hypothesis_corpus:
+            An iterable of hypothesis corpus.
+        language:
+            Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+        return_sentence_level_score:
+            An indication of whether sentence-level EED is to be returned.
+        alpha:
+            optimal jump penalty, penalty for jumps between characters
+        rho:
+            coverage cost, penalty for repetition of characters
+        deletion:
+            penalty for deletion of character
+        insertion:
+            penalty for insertion or substitution of character
+
+    Returns:
+        Extended edit distance score as a tensor
+
+    Example:
+        >>> from torchmetrics.functional import eed
+        >>> reference_corpus = ["this is the reference", "here is another one"]
+        >>> hypothesis_corpus = ["this is the prediction", "here is an other sample"]
+        >>> eed(reference_corpus=reference_corpus, hypothesis_corpus=hypothesis_corpus)
+        tensor(0.3204)
+
+    References:
+        [1] P. Stanchev, W. Wang, and H. Ney, “EED: Extended Edit Distance Measure for Machine Translation”,
+        submitted to WMT 2019. `EED`_
+    """
+    scores, total_num_sentences, sentence_eed = _eed_update(
+        reference_corpus, hypothesis_corpus, language, return_sentence_level_score, alpha, rho, deletion, insertion
+    )
+
+    if return_sentence_level_score:
+        return sentence_eed
+
+    average = _eed_compute(scores, total_num_sentences)
+    return average
```

Comment:
```suggestion
    average = _eed_compute(scores, total_num_sentences)
    if return_sentence_level_score:
        return average, sentence_eed
    return average
```

Please, handle this similar to https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/text/chrf.py#L179

Reply:
added on next commit